packages/forms/src/elements/Textarea.tsx

```diff
@@ -20,27 +22,160 @@ export interface ITextareaProps extends TextareaHTMLAttributes<HTMLTextAreaEleme
+      const border =
+        parseStyleValue(computedStyle.borderTopWidth) +
+        parseStyleValue(computedStyle.borderBottomWidth);
+
+      const innerHeight = shadowTextArea.scrollHeight - padding;
+
+      shadowTextArea.value = 'x';
+      const singleRowHeight = shadowTextArea.scrollHeight - padding;
+
+      let outerHeight = innerHeight;
+
+      if (minRows) {
+        outerHeight = Math.max(Number(minRows) * singleRowHeight, outerHeight);
+      }
+
+      if (maxRows) {
+        outerHeight = Math.min(Number(maxRows) * singleRowHeight, outerHeight);
+      }
+
+      outerHeight = Math.max(outerHeight, singleRowHeight);
+
+      const updatedHeight = outerHeight + (boxSizing === 'border-box' ? padding + border : 0);
+      const overflow = Math.abs(outerHeight - innerHeight) <= 1;
+
+      setState(prevState => {
+        if (
+          (updatedHeight > 0 && Math.abs((prevState.height || 0) - updatedHeight) > 1) ||
+          prevState.overflow !== overflow
+        ) {
+          return {
+            overflow,
+            height: updatedHeight
+          };
+        }
 
-  return <StyledTextarea {...(combinedProps as any)} />;
-});
+        return prevState;
+      });
+    }, [maxRows, minRows, textAreaRef, isAutoResizable]);
+
+    const onChangeHandler = useCallback(
+      (e: React.ChangeEvent<HTMLTextAreaElement>) => {
+        if (!isControlled) {
+          calculateHeight();
+        }
+
+        if (onChange) {
+          onChange(e);
+        }
+      },
+      [calculateHeight, isControlled, onChange]
+    );
+
+    useEffect(() => {
+      if (!isAutoResizable) {
+        return undefined;
+      }
+
+      const resizeHandler = debounce(() => {
+        calculateHeight();
+      });
+
+      window.addEventListener('resize', resizeHandler);
+
+      resizeHandler();
+
+      return () => {
+        resizeHandler.cancel();
+        window.removeEventListener('resize', resizeHandler);
+      };
+    }, [calculateHeight, isAutoResizable]);
+
+    const computedStyle: React.CSSProperties = {};
+
+    if (isAutoResizable) {
+      computedStyle.height = state.height;
+      computedStyle.overflow = state.overflow ? 'hidden' : undefined;
+    }
+
+    let combinedProps = {
+      ref: textAreaRef,
+      rows: minRows,
+      onChange: onChangeHandler,
+      style: {
+        ...computedStyle,
+        ...style
+      },
+      ...props
+    };
+
+    if (fieldContext) {
+      combinedProps = fieldContext.getInputProps(combinedProps, { isDescribed: true });
+    }
+
+    return (
+      <>
+        <StyledTextarea {...(combinedProps as any)} />
+        {isAutoResizable ? (
+          <StyledTextarea
+            aria-hidden
+            readOnly
+            isHidden
+            className={props.className}
+            ref={shadowTextAreaRef}
+            tabIndex={-1}
+            style={style}
+          />
+        ) : null}
```

Comment:
Is this more readable as `{isAutoResizable && <StyledTextarea... />}`?

Reply:
I just stumbled onto https://kentcdodds.com/blog/use-ternaries-rather-than-and-and-in-jsx and it had some good points against `&&` in render functions.

I can update this if you feel strongly against it.

Reply:
The article is guarding against array checking. As this is a boolean, I'd like to think `&&` is a reliable choice.

Reply:
Updated.